File: public/usage-examples/graphics/current_clip/current_clip-1-simple.cpp
#include "splashkit.h"

int main() 
{
    // Open a window with the title "Current Clip" and dimensions
    window window = open_window("Current Clip", 800, 600);

    // Define the clipping area
    rectangle rect = current_clip(window);

    // Draw the clipping information
    draw_text("Current Clip: " + std::to_string(rect.width) + " X " + std::to_string(rect.height), COLOR_BLACK, "Arial", 24, 100, 100);
    
    // Refresh the screen to display the text
    refresh_screen();

    // Wait for 5 seconds
    delay(5000);

    // Close all windows
    close_all_windows();

    return 0;
}

File: public/usage-examples/graphics/current_clip/current_clip-1-simple.py
from splashkit import *

# Open a window with the title "Current Clip" and dimensions of 800x600 pixels
window = open_window("Current Clip", 800, 600)
font = load_font("Arial", "arial.ttf")

# Retrieve the current clipping area
rectangle = current_clip()

# Draw the clipping area dimensions as text on the screen
draw_text("Current Clip: " + str(rectangle.width) + " X " + str(rectangle.height), color_black(), font, 24, 100, 100)

# Refresh the screen to display the text
refresh_screen()

# Wait for 5 seconds
delay(5000)

# Close all windows
close_all_windows()

